src/states/node.rs

```diff
@@ -75,6 +75,70 @@ const MERGE_TIMEOUT_SECS: u64 = 300;
 /// Duration for which to hold the bootstrappers, in seconds.
 const BOOTSTRAPPER_HOLD_DUR_SECS: u64 = 300;
 
+#[derive(Default)]
+struct UnionOurMergedSection {
+    merge_prefix: Prefix<XorName>,
+    version: u64,
+    send_other_section_merge: bool,
+    our_merged_section: BTreeSet<XorName>,
+}
+
+impl UnionOurMergedSection {
+    // If multiple nodes dropped from our section, duplicated `OwnSectionMerge` will be sent,
+    // containing different dropped out nodes. This will result in different `OtherSectionMerge` to
+    // be sent out, hence causing non-accumulation on the receiver side.
+    // So here we will take a union of our_merged_section from all the `OwnSectionMerge`, and resend
+    // `OtherSectionMerge` whenever the new union is different to the last one sent out.
+    fn union_our_merged_section(&mut self,
```

Comment:
Since that's already the type name, maybe this should just be `extend`, or `add`?